camel-idea-plugin/src/main/java/org/apache/camel/idea/preference/properties/CamelIgnoreAndExcludePage.java

```diff
@@ -0,0 +1,236 @@
+import org.apache.camel.idea.service.CamelPreferenceService;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+class CamelIgnoreAndExcludePage extends BaseConfigurable implements SearchableConfigurable, Configurable.NoScroll {
+
+    private List<String> myExcludedProperties;
+    private AddEditRemovePanel<String> excludePropertyFilePanel;
+
+    private List<String> myIgnoredProperties;
+    private AddEditRemovePanel<String> ignorePropertyFilePanel;
+
+    CamelIgnoreAndExcludePage() {
+    }
+
+    @NotNull
+    @Override
+    public String getId() {
+        return "preference.CamelConfigurable";
+    }
+
+    @Nls
+    @Override
+    public String getDisplayName() {
+        return "Ignore and exclude properties";
+    }
+
+    @Nullable
+    @Override
+    public JComponent createComponent() {
+        JPanel result = new JPanel(new BorderLayout());
+        JPanel propertyTablePanel = new JPanel(new VerticalLayout(1));
+        propertyTablePanel.add(createIgnorePropertiesFilesTable());
+        propertyTablePanel.add(createExcludePropertiesFilesTable());
+        result.add(propertyTablePanel);
+
+        return result;
+    }
+
+    @Override
+    public void apply() throws ConfigurationException {
+        CamelPreferenceService service = getCamelPreferenceService();
+        service.setExcludePropertyFiles(myExcludedProperties);
+        service.setIgnorePropertyList(myIgnoredProperties);
+
+        setModified(false);
+    }
+
+    @Override
+    public void reset() {
+        resetIgnorePropertiesTable();
+        resetExcludePropertiesTable();
+
+        setModified(false);
+    }
+
+    private void resetExcludePropertiesTable() {
+        myExcludedProperties = new ArrayList<>();
+        List<String> excludePropertyList = getCamelPreferenceService().getExcludePropertyFiles();
+        ContainerUtil.addAll(myExcludedProperties, excludePropertyList);
+        excludePropertyFilePanel.setData(myExcludedProperties);
+    }
+
+    private void resetIgnorePropertiesTable() {
+        myIgnoredProperties = new ArrayList<>();
+        List<String> ignorePropertyList = getCamelPreferenceService().getIgnorePropertyList();
+        ContainerUtil.addAll(myIgnoredProperties, ignorePropertyList);
+        ignorePropertyFilePanel.setData(myIgnoredProperties);
+    }
+
+    private CamelPreferenceService getCamelPreferenceService() {
+        return ServiceManager.getService(CamelPreferenceService.class);
+    }
+
+    private JPanel createExcludePropertiesFilesTable() {
+        final JPanel mainPanel = new JPanel(new GridLayout(1, 1));
+        mainPanel.setPreferredSize(JBUI.size(300, 300));
+        mainPanel.setBorder(BorderFactory.createEmptyBorder(5, 5, 0, 0));
+        excludePropertyFilePanel = new AddEditRemovePanel<String>(new IgnoredUrlsModel(), getCamelPreferenceService().getExcludePropertyFiles(), "Exclude property file list") {
```

Comment:
You could move this out to it's own internal class and let it extend AddEditRemovePanel and re-use it below also I guess

Reply:
@bennetelli did you see this comment?

Reply:
@fharms fixed :)